dex/src/main/java/io/crate/data/join/HashInnerJoinBatchIterator.java

```diff
@@ -48,7 +48,11 @@
  *     for (rightRow in right) {
  *         if (hash(rightRow) found in Buffer {
  *            for (row in matchedInBuffer) { // Handle duplicate values from left and hash-conflicts
- *                emmit(combinedRow)
+ *                if (joinCondition matches) {
+ *                    // We need to check that the joinCondition matches as we can have a hash-conflict
+ *                    // or the join condition can contain more operators. See HashJoinConditionSymbolsExtractor.
```

Comment:
hm not sure if mentioning the HashJoinConditionSymbolsExtractor helps here. if so using a link notation would be better, but I'd just leave this out.

Reply:
the link notation doesn't work as we don't depend on the `:sql` module.
I kind of like to have a pointer to what I mean and not re-explain it here. thoughts?

Reply:
whats about: row-lookup-by-hash-code can only work by the EQ operators of a join condition, all other possible operators must be checked afterwards.